### PR TITLE
More module tests and implement RegisterModuleSource

### LINKED ISSUE #4577: ch: Implement WScript.RegisterModuleSource(path, source)
Adding multi-file tests is hard and unnecessarily complicated comparing to `jshost`. Implement `Implement WScript.RegisterModuleSource(path, source)` so the test file doesn't have to be a separate file on the file system.

### PR DESCRIPTION
**Pull request by obastemur**
- Implement RegisterModuleSource
- es6-modules: add more test cases and move it to separate folder

Fixes #4577

### CODE REVIEW

File: bin/ch/WScriptJsrt.cpp [hunk 1]
```diff
@@ -1551,19 +1607,23 @@ HRESULT WScriptJsrt::ModuleMessage::Call(LPCSTR fileName)
 
             if (FAILED(hr))
             {
-                if (!HostConfigFlags::flags.MuteHostErrorMsgIsEnabled)
+                // check if have it registered
+                AutoString *data;
```
Comment (jackhorton) [inline bin/ch/WScriptJsrt.cpp]: Since `data` is a pointer, I don't think the AutoString destructor will run -- is that an issue?

**Comment by obastemur** [comment]
> I am pretty sure that there are memory leaks in the new AutoString/FileNode/SourceMap code. Could it be altered to use RAII and const references and fewer pointers?

Could you please point them out?

File: bin/ch/WScriptJsrt.cpp (hunk 1 above)
Comment (obastemur) [reply bin/ch/WScriptJsrt.cpp]: It shouldn't run by design and intention. See constructor. It takes over the ownership (there is a comment). Source code map lifetime == process lifetime.

**Comment by jackhorton** [comment]
So I think the part I am considering leaking is https://github.com/Microsoft/ChakraCore/pull/4582/files#diff-05783da3b8bba4fa92445ddb47da7d51R309, where we `new` without ever deleting. I suppose thats fine since the FileNodes are never removed from the linked list, but a comment explaining why that `new` was safe would be nice.

**Comment by rhuanjl** [comment]
Will the idea in time to be to move most tests out of the separate files and into the combined files with this new API?

Seems great from the perspective of making the tests easier to update and keep track of. My only concern would be if we dropped all of the separate tests the file loading routines would stop getting tested - this may be a ch only issue and not a chakracore issue but probably best to keep them intact.

**Comment by obastemur** [comment]
@rhuanjl there is no greater plan here. I just saw an opportunity to use tests from our internal testing tool and implemented the interface. ++ moved the tests into sep. folder.

I wouldn't change any of the existing tests but going forward we better use the new interface instead of adding more files.

**Pull request merged**